content/general/events/azure.textile

```diff
@@ -0,0 +1,45 @@
+title: Azure Functions
+section: general
+index: 23
+languages:
+  - none
+jump_to:
+  Help with:
+    - Fields#fields
+    - Example#example
+---
+
+As part of "Reactor Events":/general/events, it is possible to integrate with various systems.
+
+"Azure Functions":https://azure.microsoft.com/en-gb/services/functions provide event-driven serverless compute functions which allow users to easily run code whenever events are sent to it.
+
+h2(#tutorials). Tutorials & Examples
+
+If you would like to just dive in and see an example of this being implemented, then take a look at our "Azure Functions tutorial":/tutorials/reactor-event-azure.
+
+h2(#fields). Rule fields
+
+- Azure App ID := The App ID of your "Azure app":https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-azure-function#create-a-function-app
+- Function Name := the name of your "Azure Function":https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-azure-function#create-a-function-app
+- Headers := Optionally allows you to provide a set of headers that will be included in all HTTP POST requests. You must use format @"XCustom-Header-1:value1"@ for each header you add
+- "Source":/general/events#sources := Choose which of @Message@, @Presence@, or @Channel Lifecycle@ events on channels should activate this Reactor Event Rule. @Channel Lifecycle@ events are only available in "Batch Request":/general/events#batching mode
+- "Request Mode":/general/events#batching := This will either be in @Single Request@ mode or @Batch Request@ mode. "Single Request":/general/events#single-request will send each event as seperately to the endpoint specified by the Rule. "Batch Request":/general/events#batch-request will send events triggered concurrently in the same request
+- Channel filter := An optional filter, which allows the Rule to be applied to a restricted set of channels. This can be specified as a regular expression, allowing for swathes of channels to be used
+- Encoding := The encoding to be used by this Rule. This can be either JSON or "MsgPack":http://msgpack.org
+
+<br/>
+
+If the Rule is in the *Single Request* mode, it will also have the following options:
+
+- "Enveloped":/general/events#enveloping := If the rule has the Enveloped option set, then data delivered by this Rule will be wrapped in an "Ably envelope":/general/events#envelope-examples. Otherwise, the Rule will send the "raw payload":/general/events#no-envelope-examples
+
+<br/>
+
+If the Rule is in the *Batch Request* mode, it will have the following additional options:
+
+- Sign with key := Ably will optionally sign the data with the specified private key. This will be included as an HTTP header @X-Ably-Signature@ in every HTTP post request issued to your server. See "webhook security":/general/events#security for more details.
+
+h2(#example). Examples
+
+Messages to Azure Functions will match the examples provided on the "main Reactor Events page":/general/events#examples.
```

Comment:
Could we not just use partials and thus reduce the need for developers to refer back to general docs each time?